hack/generate_git_data.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+URL=$1
+
+mkdir tmp_git_dir
+cd tmp_git_dir
+
+git init -b main --quiet
+git remote add origin $URL
+git pull origin main --quiet
+git fetch --tags --quiet
```

Comment:
I'm not sure we can do this in a hermetic build and where we need all sources to be part of the SRPM...?

Why can't we simply vendor the Go dependencies, tar everything up, and inject the params via `--define` or a global? See https://github.com/openshift/microshift/blob/main/packaging/rpm/microshift.spec, for example?

Reply:
Somehow Joshua did it [here](https://pkgs.devel.redhat.com/cgit/rpms/flightctl/tree/flightctl.spec?h=rhacm-2.13-rhel-9#n29) too...